### PR TITLE
Project tokens TVL breakdown endpoint

### PR DESCRIPTION
**Pull request by sdlyy**
Resolves L2B-2008

## What's changed
- New endpoint for asset's breakdowns per projects
- Migration to improve balances table read times

External and Native assets are purely based on latest reports. Canonical are based on balances & prices, since we have to get per-holder (escrow) distinction, which is missing within the reports.

### CODE REVIEW

File: packages/backend/src/peripherals/database/migrations/071_balances_unix_timestamp_index.ts
Comment (sdlyy) [inline packages/backend/src/peripherals/database/migrations/071_balances_unix_timestamp_index.ts:1]: Index has size under 500 MB vs PK which is currently over 12GB. Migration on dump takes around 150 seconds

File: packages/backend/src/api/controllers/tvl/detailedTvl.ts [hunk 1]
```diff
@@ -85,3 +81,152 @@ export function getProjectTokensCharts(
...
+    const usdValue = asNumber(report.usdValue, 2)
+    const usdPrice = usdValue / amount
+
+    return {
+      projectId: report.projectId,
+      assetId: token.id,
+      chainId: report.chainId,
+      amount: String(amount),
+      usdValue: String(usdValue),
+      usdPrice: String(usdPrice),
```
Comment (michalsidzej) [inline packages/backend/src/api/controllers/tvl/detailedTvl.ts:125]:
```suggestion
      amount: amount.toString(),
      usdValue: usdValue.toString(),
      usdPrice: usdPrice.toString(),
```

File: packages/backend/src/api/controllers/tvl/detailedTvl.ts [hunk 2]
```diff
@@ -85,3 +81,152 @@ export function getProjectTokensCharts(
...
+        const amount = asNumber(escrowTokenBalance.balance, token.decimals)
+        const usdValue = amount * price.priceUsd
+
+        return {
+          projectId: project.projectId,
+          assetId: token.id,
+          chainId: token.chainId,
+          amount: String(amount),
+          usdValue: String(usdValue),
+          usdPrice: String(price.priceUsd),
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/backend/src/api/controllers/tvl/detailedTvl.ts]: https://github.com/l2beat/l2beat/pull/1798/commits/7aa7497ced75b180d2e674697e903d663a2d1476

File: packages/backend/src/peripherals/database/BalanceRepository.ts [hunk 3]
```diff
@@ -41,6 +41,19 @@ export class BalanceRepository extends BaseRepository {
     return rows.map(toRecord)
   }
 
+  async getByTimestampWithAnyChain(
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/backend/src/peripherals/database/BalanceRepository.ts]: https://github.com/l2beat/l2beat/pull/1798/commits/a79b374b4db1e2310a1c672229a6d2bf5503b11b

**Pull request merged**